ykcompile/src/lib.rs

```diff
@@ -208,11 +192,10 @@ impl TraceCompiler {
             }
             ykpack::CallOperand::Unknown => {}
         };
-        // Save all registers to the stack.
-        self.store_registers();
         // Move call arguments into registers.
+        let local_offset = dest.as_ref().unwrap().local;
         for (i, op) in args.iter().enumerate() {
-            let argidx = Local((i + 1) as u32);
+            let argidx = Local(i as u32 + local_offset.0 + 1);
```

Comment:
In general `as` scares me, as it can do some surprising things. It might not be possible, but perhaps there's a way of shuffling the code that means we can consistently use one integer type?

Reply:
The problem is that `Local` takes `u32`s but enumerate returns `usize`. I think I may be able to use `zip` to enumerate while using `u32` as an iterator.

Reply:
Fixed in 5dc9bb8